.github/workflows/nx-integration-tests.yml

```diff
@@ -0,0 +1,186 @@
+        with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.pr_head }}
+
+      - name: Calculated Result
+        shell: bash
+        env:
+          php_versions: ${{ steps.calculate-matrix.outputs.php_versions }}
+          database_versions: ${{ steps.calculate-matrix.outputs.database_versions }}
+          search_versions: ${{ steps.calculate-matrix.outputs.search_versions }}
+          message_queue_versions: ${{ steps.calculate-matrix.outputs.message_queue_versions }}
+          cache_versions: ${{ steps.calculate-matrix.outputs.cache_versions }}
+          http_cache_versions: ${{ steps.calculate-matrix.outputs.http_cache_versions }}
+        run: |
+          echo "PHP Versions: $php_versions"
+          echo "database_versions: $database_versions"
+          echo "search_versions: $search_versions"
+          echo "message_queue_versions: $message_queue_versions"
+          echo "cache_versions: $cache_versions"
+          echo "http_cache_versions: $http_cache_versions"
+
+  nx-project-setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run Nx Project Setup
+        uses: mage-os/github-actions/nx-integration-tests-setup@main
+        with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.pr_head }}
+          pr_base: ${{ inputs.pr_base }}
+      
+  integration-tests:
+    needs: 
+      - matrix-calculator
+      - nx-project-setup
+    strategy:
+      fail-fast: false
+      matrix:
+        php_version: ${{ fromJSON(needs.matrix-calculator.outputs.php_versions) }}
+        database_version: ${{ fromJSON(needs.matrix-calculator.outputs.database_versions) }}
+        search_version: ${{ fromJSON(needs.matrix-calculator.outputs.search_versions) }}
+        message_queue_version: ${{ fromJSON(needs.matrix-calculator.outputs.message_queue_versions) }}
+        cache_version: ${{ fromJSON(needs.matrix-calculator.outputs.cache_versions) }}
+        http_cache_version: ${{ fromJSON(needs.matrix-calculator.outputs.http_cache_versions) }}
+    runs-on: ubuntu-latest
+    steps:
+      # - name: Running With
+      #   shell: bash
+      #   run: |
+      #     echo "PHP Versions: ${{ matrix.php_version }}"
+      #     echo "database Versions: ${{ matrix.database_version }}"
+      #     echo "search Versions: ${{ matrix.search_version }}"
+      #     echo "message_queue Versions: ${{ matrix.message_queue_version }}"
+      #     echo "cache Versions: ${{ matrix.cache_version }}"
+      #     echo "http_cache Versions: ${{ matrix.http_cache_version }}"
+
+      - name: Project Cache
+        uses: actions/cache/restore@v3
+        with:
+          path: main
+          key: ${{ runner.os }}-project-${{ inputs.ref }}
+      
+      # could probably swap this to a docker container in future
+      - name: Install NX
+        working-directory: ./main
+        run: |
+          npm install -g nx@15.4.1

```

Comment:
Do we need this very specific version or should we use something like `^15.4` or even `latest`? Also applies to the other workflows.

Reply:
I know it works with that verison.
Chose to leave it as exact as I feel it's more robust that way.
We could losen to `^15.4` , but I'd be wary of `latest` as we don't want it to be flakey?
What do you think?

Reply:
sure, have removed

Reply:
IMHO, we should use `^15.4` then - I have good hope they follow semver :-P

Reply:
🤞 
this should be done now pal

Reply:
Only at one place, mind updating the other as well? :see_no_evil:

Reply:
BOOOOO! 
Haha yeah sure apologies on that

Reply:
all done